exchanges/bitget/bitget_websocket.go

```diff
@@ -0,0 +1,1212 @@
+		case bitgetTrade:
+			err = e.tradeDataHandler(&wsResponse)
+		case bitgetBookFullChannel:
+			err = e.orderbookDataHandler(&wsResponse)
+		case bitgetAccount:
+			err = e.accountUpdateDataHandler(&wsResponse, respRaw)
+		default:
+			e.Websocket.DataHandler <- websocket.UnhandledMessageWarning{Message: e.Name + websocket.UnhandledMessage + string(respRaw)}
+		}
+	default:
+		e.Websocket.DataHandler <- websocket.UnhandledMessageWarning{Message: e.Name + websocket.UnhandledMessage + string(respRaw)}
+	}
+	return err
+}
+
+// TickerDataHandler handles incoming ticker data for websockets
+func (e *Exchange) tickerDataHandler(wsResponse *WsResponse, respRaw []byte) error {
+	respAsset := itemDecoder(wsResponse.Arg.InstrumentType)
+	switch respAsset {
+	case asset.Spot:
+		var ticks []WsTickerSnapshotSpot
+		err := json.Unmarshal(wsResponse.Data, &ticks)
+		if err != nil {
+			return err
+		}
+		for i := range ticks {
+			pair, err := pairFromStringHelper(ticks[i].InstrumentID)
+			if err != nil {
+				return err
+			}
+			e.Websocket.DataHandler <- &ticker.Price{
+				Last:         ticks[i].LastPrice.Float64(),
+				High:         ticks[i].High24H.Float64(),
+				Low:          ticks[i].Low24H.Float64(),
+				Bid:          ticks[i].BidPrice.Float64(),
+				Ask:          ticks[i].AskPrice.Float64(),
+				Volume:       ticks[i].BaseVolume.Float64(),
+				QuoteVolume:  ticks[i].QuoteVolume.Float64(),
+				Open:         ticks[i].Open24H.Float64(),
+				Pair:         pair,
+				ExchangeName: e.Name,
+				AssetType:    itemDecoder(wsResponse.Arg.InstrumentType),
+				LastUpdated:  ticks[i].Timestamp.Time(),
+			}
+		}
+	case asset.Futures:
+		var ticks []WsTickerSnapshotFutures
+		err := json.Unmarshal(wsResponse.Data, &ticks)
+		if err != nil {
+			return err
+		}
+		for i := range ticks {
+			pair, err := pairFromStringHelper(ticks[i].InstrumentID)
+			if err != nil {
+				return err
+			}
+			e.Websocket.DataHandler <- &ticker.Price{
+				Last:         ticks[i].LastPrice.Float64(),
+				High:         ticks[i].High24H.Float64(),
+				Low:          ticks[i].Low24H.Float64(),
+				Bid:          ticks[i].BidPrice.Float64(),
+				Ask:          ticks[i].AskPrice.Float64(),
+				Volume:       ticks[i].BaseVolume.Float64(),
+				QuoteVolume:  ticks[i].QuoteVolume.Float64(),
+				Open:         ticks[i].Open24H.Float64(),
+				MarkPrice:    ticks[i].MarkPrice.Float64(),
+				IndexPrice:   ticks[i].IndexPrice.Float64(),
+				Pair:         pair,
+				ExchangeName: e.Name,
+				AssetType:    itemDecoder(wsResponse.Arg.InstrumentType),
+				LastUpdated:  ticks[i].Timestamp.Time(),
+			}
+		}
+	default:
+		e.Websocket.DataHandler <- websocket.UnhandledMessageWarning{Message: e.Name + websocket.UnhandledMessage + string(respRaw)}
+	}
+	return nil
+}
+
+// CandleDataHandler handles candle data, as functionality is shared between updates and snapshots
+func (e *Exchange) candleDataHandler(wsResponse *WsResponse) error {
+	var candles [][8]string
```

Comment:
```suggestion
	var candles [][8]types.Number
```

This might work; I understand the int part but its a timestamp it shouldn't overflow.